### PR TITLE
rename helper method to reflect the non-deprecated terminology

### PR DESCRIPTION
**Pull request by rboyer**
also add a state store helper function for unit tests

### CODE REVIEW

**Review by kisunji — APPROVED** [review]

**Review by markan — APPROVED** [review]

File: agent/consul/state/state_store.go [hunk 1]
```diff
@@ -203,6 +203,28 @@ func (s *Store) Snapshot() *Snapshot {
 	return &Snapshot{s, tx, idx}
 }
 
+// WalkAllTables basically lets you dump memdb generically and exists primarily
+// for very specific types of unit tests and should not be executed in
+// production code.
+func (s *Store) WalkAllTables(fn func(table string, item interface{}) bool) error {
```
Comment (dnephin) [inline agent/consul/state/state_store.go:209]: If this is used for tests, maybe this could be in a test file instead of a production file?

**Review by dnephin — COMMENTED** [review]
Thanks for making this change!

File: agent/consul/state/state_store.go (hunk 1 above)
Comment (rboyer) [reply agent/consul/state/state_store.go:209]: It's used for tests in other packages so that wouldn't work.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/495054.

**Pull request merged**